compiler/semantic/sql.go

```diff
@@ -533,29 +534,53 @@ func (a *analyzer) semGroupBy(sch *selectSchema, in []ast.Expr) []exprloc {
 func (a *analyzer) semProjection(sch *selectSchema, args []ast.AsExpr, funcs *aggfuncs) projection {
 	out := &anonSchema{}
 	sch.out = out
-	conflict := make(map[string]struct{})
+	aliases := make(map[string]struct{})
 	var proj projection
 	for _, as := range args {
 		if isStar(as) {
 			proj = append(proj, column{})
 			continue
 		}
 		col := a.semAs(sch, as, funcs)
-		//XXX check for conflict for now, but we're getting rid of this soon
-		if _, ok := conflict[col.name]; ok {
-			n := ast.Node(as.Expr)
-			if as.ID != nil {
-				n = as.ID
+		if as.ID != nil {
+			if _, ok := aliases[col.name]; ok {
+				a.error(as.ID, fmt.Errorf("duplicate alias %q", col.name))
+				continue
 			}
-			a.error(n, fmt.Errorf("%q: conflicting name in projection; try an AS clause", col.name))
+			aliases[col.name] = struct{}{}
 		}
-		conflict[col.name] = struct{}{}
 		proj = append(proj, *col)
+	}
+	// Do we have duplicates for inferred columns? If an alias and an
+	// inferred column collide the inferred name should be changed.
+	conflict := maps.Clone(aliases)
```

Comment:
Is this necessary?

Reply:
The maps.Clone is not necessary and I was going to get rid of this. The name change made the code more readable to me but am willing to get rid of that as well.

Reply:
I got rid of it